src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -253,13 +253,37 @@ private GenericInfoBucketData createGenericInfoBucketData(BucketEntity bucket) {
     }
 
     public List<CatalogObjectMetadata> listCatalogObjectsByKind(String bucketName, String kind) {
+        System.out.println("kind " + kind);
+
         findBucketByNameAndCheck(bucketName);
         List<CatalogObjectRevisionEntity> result = catalogObjectRevisionRepository.findDefaultCatalogObjectsOfKindInBucket(bucketName,
                                                                                                                            kind);
 
         return buildMetadataWithLink(result);
     }
 
+    // find catalog objects by kind and content type
+    public List<CatalogObjectMetadata> listCatalogObjectsByKindAndContentType(String bucketName, String kind,
+            String contentType) {
+        System.out.println("kind and content " + kind + "  " + contentType);
+        findBucketByNameAndCheck(bucketName);
+        List<CatalogObjectRevisionEntity> result = catalogObjectRevisionRepository.findDefaultCatalogObjectsOfKindAndContentTypeInBucket(bucketName,
+                                                                                                                                         kind,
+                                                                                                                                         contentType);
+
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)